ui/app/components/clients/current.js

```diff
@@ -11,9 +12,15 @@ export default class Current extends Component {
   });
   @tracked selectedNamespace = null;
 
-  // TODO CMB get from model
   get upgradeDate() {
-    return this.args.upgradeDate || null;
+    let keyInfoObject = this.args.model.versionHistory.keyInfo;
+    if (!keyInfoObject) {
+      return false;
+    }
+    let earliestUpgradeVersion = Object.keys(keyInfoObject)[0]; // key name here will change, but we want the first one as this indicates when they upgraded to 1.9 or above
+    let versionDate = new Date(keyInfoObject[earliestUpgradeVersion].timestamp_installed);
+    // compare against this month and this year to show message or not.
+    return isAfter(versionDate, startOfMonth(new Date())) ? versionDate : false;
```

Comment:
Based on the method name I would expect this to return a date, not a boolean. I wonder if the naming could be updated to be more clear about that, or always return the version date and do the `isAfter` check in the consuming method